docs/learning/tutorials/develop-agent.md

```diff
@@ -2,43 +2,47 @@
+ > Allow some time after registering the agent application. The identity profile of the agent is being created in the background and can take between 5 and 10 minutes
 
 
 ### Build
 - Open the cloned repo in VS Code
 - Open integrated terminal
-- `node ./common/scripts/install-run-rush install`
-- Edit agent-app/query-agent/src/QueryAgentConfig.ts
-    > `imjs_agent_imodel_name` = The name of your iModel<br/>
-`imjs_agent_client_id` = client_id of the registered agent application<br/>
-`imjs_agent_client_secret` = The client secret of registered agent application<br/>
-
+- `npm install`
+- Create a `.env` file at the project root with the following:
+    ``` ps
+    ###############################################################################
+    # This file contains secrets - don't commit or share it!
+    ###############################################################################
 
-- `node ./common/scripts/install-run-rush build -t imodel-query-agent`
+    # Specify an iModel
+    CONTEXT_ID=
+    IMODEL_ID=
 
-### Run
-- `cd agent-app\query-agent`
-- `npm run start`
-- The agent will listen for changesets for 40 seconds
+    # OIDC configuration
+    # Don't forget to add <CLIENT_ID>@apps.imsoidc.bentley.com to your CONNECT project. This can be done in the iModel Registration Dashboard.
+    CLIENT_ID=
+    CLIENT_SECRET=
+    ```
 
-In order to see changesets being logged, you can use the [imodel-changeset-test-utility](https://github.com/imodeljs/imodeljs-samples/tree/master/tools/imodel-changeset-test-utility).
+    > The values for `CONTEXT_ID` and `IMODEL_ID` can be obtained from the IDs column of the [iModel registration dashboard](../../../getting-started/registration-dashboard/?tab=1)
 
-In a second terminal (`Ctrl+Shift+5` in VS Code's integrated terminal)
-- `cd tools\imodel-changeset-test-utility`
-- Edit tools\imodel-changeset-test-utility\src\ChangesetGenerationConfig.ts
-    > `imjs_agent_imodel_name` = The name of your iModel<br/>
-`imjs_agent_client_id` = client_id of the registered agent application<br/>
-`imjs_agent_client_secret` = cThe client secret of registered agent application<br/>
+    > The values for `CLIENT_ID` and `CLIENT_SECRET` come from the Agent Application you registered during the Setup step
 
-- `npm install`
 - `npm run build`
-- While the query-agent is running and listening for changesets...
-- `npm run start`
 
-You should see the query-agent receiving events and logging them to \agent-app\query-agent\lib\output\changeSummaries.
+### Run
+- `npm start`
+- The agent will listen for changesets pushed to iModelHub
+- For testing, it can often also be useful to skip the event listening and just run against a specific changeset. To do that, either run `npm start -- --latest` to use the latest changeset, or `npm start -- --changeset=<CHANGESETID>` to use any specific changeset
+

```

Comment:
Is there any replacement for the changeset utility?  Maybe we should point people to the synchronizer as a way to generate changesets?

Reply:
Agreed, I'd rather have that so people could use some real data to test their agent with.  I think that's what was used for the Jumpstart event when showcasing this starter.

Reply:
Added note about iTwin Synchronizer